whylog/teacher/constraint_links_base.py

```diff
@@ -1,71 +1,59 @@
+from collections import namedtuple
+
+ConstraintLink = namedtuple('ConstraintLink', ['line_id', 'group_no', 'constraint_id'])
+
+
 class ConstraintLinksBase:
     """
     Container for links between group and constraint
-    """
 
-    LINE_ID_POSITION = 0
-    GROUP_NUMBER_POSITION = 1
-    CONSTRAINT_ID_POSITION = 2
+    :param links: list of links between group and constraint
+    :type links: list[ConstraintLink]
+    """
 
-    def __init__(self, links_list=None):
+    def __init__(self, links=None):
         """
-        :param links_list: list of links between group and constraint
-                          one link is a tuple (line_id, group_no, constraint_id)
-        :type links_list: list[(int, int, int)]
+        :param links: list of links between group and constraint represented
+                      by simple tuple (line_id, group_no, constraint_id)
```

Comment:
Well, It is consistent with implementation. Constructor takes list of simple tuples as shown in tests.
Is it better to communicate with outer world through simple tuples or ConstrainLinks? Now, in methods: __init__, add_list, get_links we communicate through list of simple tuples.